### PR TITLE
Natural Language formatter class

### LINKED ISSUE #35: Refactor 'nl' logic in OpenAIHandler to be independent of backing llm provider
the 'nl' stuff can be refactored so we have a class that just handles the text formatting logic that can be used across different LLM services

_Originally posted by @jobez in https://github.com/The-Overlore/lore-machine/pull/29#discussion_r1464070829_

### LINKED ISSUE #41: bug: check katana url is set
Otherwise raise Runtime Error

### PR DESCRIPTION
**Pull request by cwastche**
resolve #35 
resolve #41

### CODE REVIEW

File: overlore/config.py [hunk 1]
```diff
@@ -50,14 +50,20 @@ def load(self) -> None:
         dotenv_path = ".env.production" if self.prod is True else ".env.development"
         load_dotenv(dotenv_path=dotenv_path)
 
-        self.TORII_WS = str(os.environ.get("TORII_WS"))
-        self.KATANA_URL = str(os.environ.get("KATANA_URL"))
-        self.TORII_GRAPHQL = str(os.environ.get("TORII_GRAPHQL"))
+        tmp_TORII_WS = os.environ.get("TORII_WS")
```
Comment (edisontim) [inline overlore/config.py:53]: lowercase everything, upper case is for "global constants" usually

**Review by edisontim — CHANGES_REQUESTED** [review]
one small comment, otherwise lgtm

File: overlore/config.py (hunk 1 above)
Comment (cwastche) [reply overlore/config.py]: Noted, and done

File: overlore/config.py [hunk 2]
```diff
@@ -44,20 +51,29 @@ def _get_args(self) -> None:
...
+        invalid_urls = [name for name, url in required_urls.items() if not self._is_valid_url(url)]
+
+        if invalid_urls:
+            raise RuntimeError(f"Invalid or missing URL(s) in .env file for: {', '.join(invalid_urls)}")
+
+        self.TORII_WS = str(tmp_torii_ws)
+        self.TORII_GRAPHQL = str(tmp_torii_graphql)
+        self.KATANA_URL = str(tmp_katana_url)
+
+        print(self.TORII_GRAPHQL)
```
Comment (edisontim) [inline overlore/config.py:77]: remove prints

File: overlore/config.py [hunk 3]
```diff
@@ -44,20 +51,29 @@ def _get_args(self) -> None:
...
         )
 
-        if self.TORII_WS is None:
-            raise RuntimeError("Failure to provide WS url")
-        if self.TORII_GRAPHQL is None:
-            raise RuntimeError("Failure to provide graphql url")
+        required_urls = {"TORII_WS": tmp_torii_ws, "TORII_GRAPHQL": tmp_torii_graphql, "KATANA_URL": tmp_katana_url}
+        invalid_urls = [name for name, url in required_urls.items() if not self._is_valid_url(url)]
+
+        if invalid_urls:
```
Comment (edisontim) [inline overlore/config.py:70]: Have just || for each `tmp_` variable

**Review by edisontim — CHANGES_REQUESTED** [review]

**Review by edisontim — APPROVED** [review]

**Pull request merged**